instrumentation/reactor/reactor-kafka-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/reactor/kafka/v1_0/DefaultKafkaReceiverInstrumentation.java

```diff
@@ -33,11 +34,14 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class CreateConsumerFluxAdvice {
 
+    @AssignReturned.ToReturned
     @Advice.OnMethodExit(suppress = Throwable.class)
-    public static void onExit(@Advice.Return(readOnly = false) Flux<?> flux) {
+    public static Flux<?> onExit(@Advice.Return Flux<?> originalFlux) {
+      Flux<?> flux = originalFlux;
       if (!(flux instanceof TracingDisablingKafkaFlux)) {
         flux = new TracingDisablingKafkaFlux<>(flux);
       }
+      return flux;
```

Comment:
Perhaps this and ConsumerHandlerInstrumentation could use the same approach as KafkaReceiverInstrumentation?

```suggestion
      if (!(flux instanceof TracingDisablingKafkaFlux)) {
        return new TracingDisablingKafkaFlux<>(flux);
      }
      return flux;
```

Reply:
https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/14640/commits/dc304625317877cfad29f625b96c9396221978e5 should cover this and also removes systematic negation from if statement.